src/item_action.cpp

```diff
@@ -120,9 +129,39 @@ item_action_map map_actions_to_items( player &p )
     return candidates;
 }
 
+const std::string &item_action_generator::get_action_name( const item_action_id &id ) const
+{
+    return get_action( id ).name;
+}
+
+const std::string &item_action_generator::get_action_name( const use_function &fun ) const
+{
+    static const std::string errstring("ERROR");
+    const auto *actor = fun.get_actor_ptr();
+    if( actor == nullptr ) {
+        debugmsg( "Multiple use_function can only have iuse_actor actions!" );
+        return errstring;
+    }
+
+    return get_action_name( actor->type );
+}
+
+const item_action &item_action_generator::get_action( const item_action_id &id ) const
+{
+    const auto &iter = item_actions.find( id );
```

Comment:
Nitpicking: the reference works and is probably completely valid, but it's confusing, why not make it a regular variable?